torch/distributions/lowrank_multivariate_normal.py

```diff
@@ -0,0 +1,176 @@
+import math
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.multivariate_normal import (_batch_diag, _batch_mahalanobis, _batch_mv,
+                                                     _batch_potrf_lower, _batch_trtrs_lower,
+                                                     _get_batch_shape)
+from torch.distributions.utils import lazy_property
+
+
+def _batch_vector_diag(bvec):
+    """
+    Returns the diagonal matrices of a batch of vectors.
+    """
+    n = bvec.shape[-1]
+    flat_bvec = bvec.reshape(-1, n)
+    flat_bmat = torch.stack([v.diag() for v in flat_bvec])
+    return flat_bmat.reshape(bvec.shape + (n,))
+
+
+def _batch_capacitance_tril(W, D):
+    r"""
```

Comment:
You don't really need to make this a doc-string, since this will not be exposed. If you want to keep it, it's fine.